### ISSUE TITLE
validate_quantities has unhelpful warning for `u.dimensionless_unscaled`

### ISSUE THREAD

**Issue by StanczakDominik** [PlasmaPy/PlasmaPy#888]
If you apply @validate_quantities on something like this:
```python
@validate_quantities
def solve_RH(gamma: u.dimensionless_unscaled):
	return gamma
```
You get the mildly unhelpful warning:
```
WARNING: UnitsWarning: The argument 'gamma' to function solve_RH() has no specified units. Assuming units of . To silence this warning, explicitly pass in an astropy Quantity (e.g. 5. * astropy.units.cm) (see http://docs.astropy.org/en/stable/units/) [plasmapy.utils.decorators.validators]
```

Note the `assuming units of .` part. This probably deserves a special case. How should we reformulate this warning in that case?

**Comment by StanczakDominik** [comment]
I honestly dislike this warning enough, and find `u.dimensionless_unscaled` cumbersome enough that I think it's fine to completely skip the warning. It's just a number.

**Comment by StanczakDominik** [comment]
cc @rocco8773

**Comment by rocco8773** [comment]
Yeah, that weird warning is because the string representation of `u.dimensionless_unscaled` is `''`.

I'm torn on this.  On one hand I don't like making exceptions to the rule since it make behavior unpredictable.  On the other hand, the unit of an `int` or `float` is basically `dimensionless_unscaled`, so `5 * u.dimensionless_unscaled` does not do much beyond bringing the variable into the astropy `Quantity` framework.

Since I don't think `val = 5` or `val = 5 * u.dimensionless_unscaled` would be treated any differently within the function, I won't oppose letting the application of `u.dimensionless_unscaled` go un-warned.
